lighthouse-core/report/html/renderer/category-renderer.js

```diff
@@ -344,7 +357,9 @@ class CategoryRenderer {
       return scoreDisplayMode;
     }
 
-    if (Util.showAsPassed(auditRef.result)) {
+    if (this._auditHasWarning(auditRef)) {
+      return 'warning';
+    } else if (Util.showAsPassed(auditRef.result)) {
```

Comment:
hmm, we should figure out what we want the priority order to be here. Does `warning` win over `failed`?

Reply:
changed to require a passing state for all warnings.